spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -2011,6 +2011,40 @@ public void testZOrderRewriteWithSpecificOutputSpecId() {
     shouldRewriteDataFilesWithPartitionSpec(table, outputSpecId);
   }
 
+  @TestTemplate
+  public void testRewriteMaxFilesOption() {
+    Table table = createTablePartitioned(5, 4);
+    shouldHaveFiles(table, 20);
+    int desiredMaxFileCount = 5000;
+    Result result =
+        actions()
+            .rewriteDataFiles(table)
+            .option(RewriteDataFiles.MAX_FILES_TO_REWRITE, String.valueOf(desiredMaxFileCount))
+            .execute();
+    table.refresh();
+    int filesReWritten = result.rewrittenDataFilesCount();
+    assertThat(filesReWritten).isLessThanOrEqualTo(desiredMaxFileCount);
+  }
+
+  @TestTemplate
+  public void testRewriteMaxFilesOptionEquality() {
+    Table table = createTablePartitioned(5, 4);
+    writeRecords(1, SCALE, 1);
+    writeRecords(2, SCALE, 2);
+    writeRecords(3, SCALE, 3);
+    writeRecords(4, SCALE, 4);
+    shouldHaveFiles(table, 50);
+    int desiredMaxFileCount = 10;
+    Result result =
+        actions()
+            .rewriteDataFiles(table)
+            .option(RewriteDataFiles.MAX_FILES_TO_REWRITE, "10")
+            .execute();
+    table.refresh();
+    int filesReWritten = result.rewrittenDataFilesCount();
+    assertThat(filesReWritten).isEqualTo(desiredMaxFileCount);
+  }
+
```

Comment:
I would add one test to our validation test (somewhere in this file) to test that a - or 0 value for the new param errors out as expected

Reply:
We are also missing a test where max files to rewrite causes any early termination within a partition I think. For example a table with a single partition with 7 files and max_files_to_rewrite = 10. I think the code above for subList will break here since the index I think has to be within the list itself? I haven't checked

Reply:
Thank you for the feedback . I added a couple more tests (to address the max files rewrite > total files in a partitions inline with your example above) and also to verify the right exception along with error message are thrown in case an invalid value is passed